content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -171,7 +171,7 @@ how to manage these tokens with `kubeadm`.
 A service account is an automatically enabled authenticator that uses signed
 bearer tokens to verify requests. The plugin takes two optional flags:
 
-* `--service-account-key-file` A file containing a PEM encoded key for signing bearer tokens.
+* `--service-account-key-file` A file containing a PEM encoded key for verifying bearer tokens.
 If unspecified, the API server's TLS private key will be used.
```

Comment:
If it's only verifying not signing, using private key as the default is not so reasonable for me.

(I'm not familiar to this topic, just being confused.)

Reply:
Yeah, it is confusing. The default is not great.

It works because the public key is embedded in the private key PEM file.

Reply:
How about this sentence? (not sure if it's correct)

> If unspecified, the signing key will be derivative from API server's TLS private key.

Reply:
I just want to fix the blatant error. Someone can rewrite the section with more information in a follow-up PR.

But the above suggestion is not correct. It's not a signing key but a key used for verification. Also, ["key derivation" has special meaning in cryptography](https://en.wikipedia.org/wiki/Key_derivation_function) which is not what's happening here.

Reply:
copying the flag's actual help text is probably most correct and consistent:

> File containing PEM-encoded x509 RSA or ECDSA private or public keys, used to verify ServiceAccount tokens. The specified file can contain multiple keys, and the flag can be specified multiple times with different files. If unspecified, --tls-private-key-file is used. Must be specified when --service-account-signing-key-file is provided.

Reply:
Done. I skipped the last line. We can add that later if we add information about the other flags on this page.

Reply:
> It works because the public key is embedded in the private key PEM file.

Aside: all the kinds of private key that you can put in a `.pem` file also offer you a way to derive the public key just from the private part.

If adopting the command line parameter's help text, consider light paraphrasing to fit the website [style guide](https://kubernetes.io/docs/contribute/style/style-guide/).